ironfish/src/wallet/wallet.ts

```diff
@@ -229,6 +229,16 @@ export class Wallet {
     }
     this.isStarted = true
 
+    for (const account of this.listAccounts()) {
+      if (account.createdAt === null) {
+        continue
+      }
+
+      if (!(await this.chain.hasBlock(account.createdAt.hash))) {
```

Comment:
What if we import this into an wallet that hasn't synced the chain, it'll clear the createdAt even if the block will exist later?

Reply:
It will clear createdAt in this case since we can't know for sure if the block will exist later. 

We could check `createdAt.sequence` to check whether createdAt might be ahead of the chain an only clear createdAt if its sequence is behind the current head and the block isn't on the chain.

If we do that I think we'll also need to update the scan logic to check whether createdAt is on the main chain when we get to that sequence and reset if it isn't.

Reply:
That sounds good, but is there any danger to just letting it stay as a block that's not on the same chain?

Is it that you might skip blocks between a fork point and the current head when you should check those for transactions?

Reply:
> Is it that you might skip blocks between a fork point and the current head when you should check those for transactions?

Yeah exactly, if the account was created on a fork or on another chain (e.g., testnet), then scanning might skip blocks that we should check.